### PR TITLE
Add entry points for supported inventory sources

### PR DESCRIPTION
**Pull request by jessicamack**
A related change for https://github.com/ansible/awx/pull/15658

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -89,6 +89,12 @@ insights = "awx_plugins.inventory.plugins:insights"
 openshift_virtualization = "awx_plugins.inventory.plugins:openshift_virtualization"
 constructed = "awx_plugins.inventory.plugins:constructed"
 
+[project.entry-points."awx_plugins.inventory.supported"]  # new entry points group name
```
Comment (webknjaz) [inline pyproject.toml]:
```suggestion
[project.entry-points."awx_plugins.inventory.supported"]  # plugins that awx will skip activating automatically
```

File: pyproject.toml [hunk 2]
```diff
@@ -76,7 +76,7 @@ thycotic_tss = "awx_plugins.credentials.tss:tss_plugin"
 aws_secretsmanager_credential = "awx_plugins.credentials.aws_secretsmanager:aws_secretmanager_plugin"
 
 [project.entry-points."awx_plugins.inventory"]  # new entry points group name
-azure-rm = "awx_plugins.inventory.plugins:azure_rm"
+azure_rm = "awx_plugins.inventory.plugins:azure_rm"
```
Comment (webknjaz) [inline pyproject.toml:79]: The tests must be adjusted for this, so they don't fail.

File: pyproject.toml [hunk 3]
```diff
@@ -89,6 +89,12 @@ insights = "awx_plugins.inventory.plugins:insights"
 openshift_virtualization = "awx_plugins.inventory.plugins:openshift_virtualization"
 constructed = "awx_plugins.inventory.plugins:constructed"
 
+[project.entry-points."awx_plugins.inventory.supported"]  # new entry points group name
+satellite6 = "awx_plugins.inventory.plugins:satellite6_supported"
+controller = "awx_plugins.inventory.plugins:controller_supported"
+insights = "awx_plugins.inventory.plugins:insights_supported"
+openshift_virtualization = "awx_plugins.inventory.plugins:openshift_virtualization_supported"
```
Comment (webknjaz) [inline pyproject.toml:97]: These should probably be mentioned in the tests too.

File: pyproject.toml (hunk 2 above)
Comment (jessicamack) [reply pyproject.toml:79]: Which tests? When I point the requirements for awx at this branch on my repo the related tests pass. I'll run them again to confirm.

Comment (webknjaz) [reply pyproject.toml:79]: Ones failing in CI — smoke: https://github.com/ansible/awx-plugins/blob/8e8d03b/tests/importable_test.py#L76 / https://github.com/ansible/awx-plugins/actions/runs/12121157958/job/33791551145?pr=61#step:17:51

File: pyproject.toml (hunk 3 above)
Comment (jessicamack) [reply pyproject.toml:97]: Added a test to cover the supported inventory plugins

File: tests/importable_test.py [hunk 4]
```diff
@@ -136,6 +136,29 @@ def __str__(self) -> str:
     ),
 )
 
+supported_inventory_plugins = (
```
Comment (webknjaz) [inline tests/importable_test.py]: Merge them. From the runtime perspective, they are the same and there shouldn't be any difference in this.
```suggestion
```

File: tests/importable_test.py [hunk 5]
```diff
@@ -192,3 +221,14 @@ def test_entry_points_are_inventory_plugin(
...
+
+@with_supported_inventory_plugins
+def test_entry_points_are_supported_inventory_plugin(
+        entry_point: EntryPointParam,
+) -> None:
+    """Ensure all exposed inventory plugins are of the same class."""  # noqa: D200, DAR101; FIXME
+    entry_points = _discover_entry_points(group=entry_point.group)
+    loaded_plugin_class = entry_points[entry_point.name].load()
+
+    assert issubclass(loaded_plugin_class, PluginFileInjector)
```
Comment (webknjaz) [inline tests/importable_test.py]:
```suggestion
```

File: tests/importable_test.py [hunk 6]
```diff
@@ -150,10 +173,16 @@ def __str__(self) -> str:
...
+    'entry_point',
+    supported_inventory_plugins,
+    ids=str,
+)
+
 
 with_all_plugins = pytest.mark.parametrize(
     'entry_point',
-    credential_plugins + inventory_plugins,
+    credential_plugins + inventory_plugins + supported_inventory_plugins,
```
Comment (webknjaz) [inline tests/importable_test.py]:
```suggestion
    credential_plugins + inventory_plugins,
```

File: tests/importable_test.py [hunk 7]
```diff
@@ -150,10 +173,16 @@ def __str__(self) -> str:
     ids=str,
 )
 
+with_supported_inventory_plugins = pytest.mark.parametrize(
+    'entry_point',
+    supported_inventory_plugins,
+    ids=str,
+)
+
```
Comment (webknjaz) [inline tests/importable_test.py]:
```suggestion
```

**Review by webknjaz — COMMENTED** [review]
I suggest simplifying the tests like this.

**Review by webknjaz — APPROVED** [review]

**Review by chrismeyersfsu — APPROVED** [review]

**Pull request merged**